internal/pkg/dns/dns.go

```diff
@@ -25,6 +25,25 @@ func (s *Server) ServeDNS(w dns.ResponseWriter, r *dns.Msg) {
 	domain := strings.ToLower(r.Question[0].Name)
 	switch r.Question[0].Qtype {
 	case dns.TypeA:
+		if domain == "localhost." {
+			// BUG(lukeshu): I have no idea why a lookup
+			// for localhost even makes it to here on my
+			// home WiFi when connecting to a k3sctl
+			// cluster (but not a kubernaut.io cluster).
+			// But it does, so I need this in order to be
+			// productive at home.  We should really
+			// root-cause this, because it's weird.
+			msg := dns.Msg{}
+			msg.SetReply(r)
+			msg.Authoritative = true
+			msg.RecursionAvailable = true
+			msg.Answer = append(msg.Answer, &dns.A{
+				Hdr: dns.RR_Header{Name: r.Question[0].Name, Rrtype: dns.TypeA, Class: dns.ClassINET, Ttl: 60},
+				A:   net.IPv4(127, 0, 0, 1),
+			})
+			_ = w.WriteMsg(&msg)
+			return
+		}
```

Comment:
I understand the need for the hack, but why does this need to be a copy-paste of the code below? Why not just set/replace `ip` and let a single code path handle it?